### PR TITLE
dev-python/tpm2-pytss: Bump to 2.2.1, remove old

### PR DESCRIPTION
**Pull request by salahcoronya**

### CODE REVIEW

File: dev-python/tpm2-pytss/tpm2-pytss-2.2.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+LICENSE="BSD-2"
+SLOT="0"
+KEYWORDS="~amd64"
+
+IUSE="+fapi test"
+
+RDEPEND="${PYTHON_DEPS}
+	app-crypt/tpm2-tss:=[fapi=]
+	fapi? ( >=app-crypt/tpm2-tss-3.0.3:= )
```
Comment (mgorny) [inline dev-python/tpm2-pytss/tpm2-pytss-2.2.1.ebuild]: These two are `DEPEND` material indeed…

File: dev-python/tpm2-pytss/tpm2-pytss-2.2.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	dev-python/cffi[${PYTHON_USEDEP}]
+	dev-python/asn1crypto[${PYTHON_USEDEP}]
+	dev-python/cryptography[${PYTHON_USEDEP}]
+	dev-python/pycparser[${PYTHON_USEDEP}]
+	dev-python/pyyaml[${PYTHON_USEDEP}]"
```
Comment (mgorny) [inline dev-python/tpm2-pytss/tpm2-pytss-2.2.1.ebuild]: …but these are not.

i.e. my suggestion would be to start with DEPEND, then do `RDEPEND=${DEPEND} ...`.

File: dev-python/tpm2-pytss/tpm2-pytss-2.2.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+DEPEND="${RDEPEND}
+	test? ( app-crypt/swtpm )"
+
+BDEPEND="${PYTHON_DEPS}
```
Comment (mgorny) [inline dev-python/tpm2-pytss/tpm2-pytss-2.2.1.ebuild]: PYTHON_DEPS redundant here too.

File: dev-python/tpm2-pytss/tpm2-pytss-2.2.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+	dev-python/setuptools-scm[${PYTHON_USEDEP}]
+	dev-python/pkgconfig[${PYTHON_USEDEP}]"
```
Comment (mgorny) [inline dev-python/tpm2-pytss/tpm2-pytss-2.2.1.ebuild]: Sort.

**Review by mgorny — CHANGES_REQUESTED** [review]

**Comment by salahcoronya** [comment]
All the specified changes were made

**Comment by mgorny** [comment]
I'm going to merge it, assuming the tests work for you. However, they fail for me:

```pytb
FAILED test/test_crypto.py::CryptoTest::test_ecc_bad_curves - AttributeError: 'TPMS_CAPABILITY_DATA' object has no attribute '_cdata'
FAILED test/test_crypto.py::CryptoTest::test_encrypted_key - AttributeError: 'TPMS_CAPABILITY_DATA' object has no attribute '_cdata'
FAILED test/test_crypto.py::CryptoTest::test_from_pem_with_scheme - AttributeError: 'TPMS_CAPABILITY_DATA' object has no attribute '_cd
ata'
FAILED test/test_crypto.py::CryptoTest::test_from_pem_with_symmetric - AttributeError: 'TPMS_CAPABILITY_DATA' object has no attribute '_cdata'
FAILED test/test_crypto.py::CryptoTest::test_get_alg - AttributeError: 'TPMS_CAPABILITY_DATA' object has no attribute '_cdata'
FAILED test/test_crypto.py::CryptoTest::test_kdfa - AttributeError: 'TPMS_CAPABILITY_DATA' object has no attribute '_cdata'
...
```

Log: [dev-python:tpm2-pytss-2.2.1:20240308-173437.txt](https://github.com/gentoo/gentoo/files/14541398/dev-python.tpm2-pytss-2.2.1.20240308-173437.txt)

**Pull request closed without merging**